jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py

```diff
@@ -323,24 +317,34 @@ def search(
             'traversal_paths', self.default_traversal_paths
         )
 
+        # expand topk number guarantee to return topk results
+        # TODO WARNING: maybe this would degrade the query speed
+        expand_topk = top_k + self.delted_count
+
         query_docs = docs.traverse_flat(traversal_paths)
 
-        vecs = np.array(query_docs.get_attributes('embedding'))
+        vecs = np.array(query_docs.get_attributes('embedding'), dtype=np.float32)
```

Comment:
Can we assume dtype is always float32?

Reply:
I don't think so. The embeddings resulted from most of encoder executors would be `float64`

Reply:
So then this line is problematic, no?

Reply:
No, `np.array()` would perform the casting if necessary

Reply:
No, I was thinking if casting to other precision levels would affect results if the original index is built on another precision. Tested it a bit locally and it doesn't seem to affect the tests

Reply:
>  the original index is built on another precision
No, this would not happen. `faiss` only support `float32`

Reply:
> The embeddings resulted from most of encoder executors would be float64

Why? All machine learning models are trained in `float32`. Is there some conversion that happens in jina somewher?
In any case, I support casting to float32

Reply:
@tadejsv yes, you are right. All of the deep learning frameworks use `float32` as default type (except `numpy`). I provided a mistaken comment.